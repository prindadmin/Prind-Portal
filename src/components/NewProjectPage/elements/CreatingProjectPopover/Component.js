import React, { Component } from 'react'
//import PropTypes from 'prop-types'

import PopOverHandler from '../../../common/popOverHandler'

import {
  Spinner,
  Intent,
} from '@blueprintjs/core'

import * as strings from '../../../../data/Strings'

export class Element extends Component {
  static propTypes = {
  }

  getContent = () => {
    return (
      <React.Fragment>
        {strings.PROJECT_CREATION_BILLING_WARNING}
        <Spinner size='150' intent={Intent.DANGER}/>
      </React.Fragment>
    )
  }

  render () {
    return (
      <PopOverHandler>
        <div id='popup-greyer' onClick={(e) => e.stopPropagation()}>
          <div id='project-creation-popover'>
            <div id='popup-box'>
              <div className='project-creation-popover-container' onClick={(e) => e.stopPropagation()}>
                <div className='element-title'>
                  {strings.PROJECT_CREATION_IN_PROGRESS}
                </div>
                <div className='element-description'>
                  {this.getContent()}
                </div>
              </div>
            </div>
          </div>
        </div>
      </PopOverHandler>
    )
  }

}

export default Element
