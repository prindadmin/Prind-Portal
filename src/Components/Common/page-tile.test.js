
import Component from './page-tile'
import { shallow } from 'enzyme';

const mockHistoryPush = jest.fn()

var props

beforeEach(() => {
  props = {
    pageName: "Design",
    selected: false,
    linkTo: "/design",
    history: {
      push: mockHistoryPush
    }
  }
});

it('Should render not selected', () => {
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});

it('Should render selected', () => {
  props.selected = true
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});

it('Should render and push history to new page', () => {
  const component = shallow(<Component {...props} />);
  component.simulate('click')
  expect(mockHistoryPush).toHaveBeenCalledWith(props.linkTo)
});
