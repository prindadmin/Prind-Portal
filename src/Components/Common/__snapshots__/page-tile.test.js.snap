// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render not selected 1`] = `
<Blueprint3.Callout
  className="list-item-container"
  onClick={[Function]}
>
  <div
    className="details"
  >
    <h2>
      Design
    </h2>
  </div>
</Blueprint3.Callout>
`;

exports[`Should render selected 1`] = `
<Blueprint3.Callout
  className="list-item-container selected"
  onClick={[Function]}
>
  <div
    className="details"
  >
    <h2>
      Design
    </h2>
  </div>
</Blueprint3.Callout>
`;
